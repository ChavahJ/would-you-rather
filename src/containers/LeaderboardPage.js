// The Leaderboard is available at/leaderboard.
// Each entry on the leaderboard contains the following:
// the user’s name;
// the user’s avatar;
// the number of questions the user asked; and
// the number of questions the user answered.
// Users are ordered in descending order based on the sum of the number of questions they’ve answered and the number of questions they’ve asked.

const LeaderBoardPage = () => {
  return (
    <section className="leaderboard">
      <table>
        <tr>
          <th>Users</th>
          <th>Answered</th>
          <th>Created</th>
        </tr>
        <tr>
          <td>User Avatar | User Name</td>
          <td>Number of Answered Polls</td>
          <td>Number of Polls Created</td>
        </tr>
        <tr>
          <td>User Avatar | User Name</td>
          <td>Number of Answered Polls</td>
          <td>Number of Polls Created</td>
        </tr>
      </table>
    </section>
  );
};

export default LeaderBoardPage;
