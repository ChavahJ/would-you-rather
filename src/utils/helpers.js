import { useParams } from "react-router-dom";

export function formatDate(timestamp) {
  const d = new Date(timestamp);
  const time = d.toLocaleTimeString("en-US");
  return time.substr(0, 5) + time.slice(-2) + " | " + d.toLocaleDateString();
}

export const withRouter = (Component) => {
  const ComponentWithRouterProp = (props) => {
    let params = useParams();
    return <Component {...props} router={{ params }} />;
  };
  return ComponentWithRouterProp;
};
