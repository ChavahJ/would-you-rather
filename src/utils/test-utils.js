import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import { handleInitialData } from "../actions/shared";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = handleInitialData(preloadedState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
