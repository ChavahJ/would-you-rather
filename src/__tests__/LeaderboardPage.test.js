import { renderWithProviders } from "../utils/test-utils";
import { MemoryRouter } from "react-router-dom";
import LeaderboardPage from "../containers/LeaderboardPage";

it("renders correctly", () => {
  const component = renderWithProviders(
    <MemoryRouter>
      <LeaderboardPage />
    </MemoryRouter>
  );
  expect(component).toMatchSnapshot();
});
