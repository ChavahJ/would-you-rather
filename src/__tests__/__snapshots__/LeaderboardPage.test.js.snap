// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="leaderboard container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h1
              class="mt-3"
            >
              Leaderboard
            </h1>
            <table
              class="my-4 table table-striped table-bordered table-hover"
            >
              <thead>
                <tr>
                  <th>
                    Users
                  </th>
                  <th>
                    Answered
                  </th>
                  <th>
                    Created
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="leaderboard container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h1
            class="mt-3"
          >
            Leaderboard
          </h1>
          <table
            class="my-4 table table-striped table-bordered table-hover"
          >
            <thead>
              <tr>
                <th>
                  Users
                </th>
                <th>
                  Answered
                </th>
                <th>
                  Created
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
